Metro-Records/la-metro-councilmatic
# Allow for alert editing and line breaks

## Overview

This branch adds the ability to edit existing alerts within the alert manager. It also allows for line breaks to be added to alerts using `<br>` tags, and supplies a button to the editor do so.

- Connects #1076 

### Demo
![edit-alerts](https://github.com/Metro-Records/la-metro-councilmatic/assets/114717958/e6db703c-3e86-4871-991c-c56d1876537f)

### Notes

There's an adjustment to the for loop that renders alerts, that makes sure the alerts stay in the same orde when edited. Without that `dictsort` filter, newly edited alerts may appear at the top of the list.

Let me know what you think about the UI/UX for this process. The alert manager was initially kept on the same page as the index for ease of use, so I tried to keep that pattern going with this new editor. The other option I could think of was to change the existing "add new" form to an "edit" form with some jquery when the edit button was clicked. We could also just move all of this to a separate page for simplicity's sake.

Also, because of the way the view is set up, I had trouble getting backend validation to make sure alerts don't have empty descriptions. If there's a way to use the AlertForm I previously created, that would be ideal, since that already has the validation. Otherwise I could change the Alert model to remove `null=True, blank=True` from the description.

## Testing Instructions

 * Login to the app
 * Create an alert with some line breaks using the editor button or by typing in the break tag `<br>`
   * Confirm that line breaks appear in the alert
 * Use the alert manager to edit the description and type, then save.
   * Confirm that the alert is changed
 * Optional: create/edit alerts and confirm they stay in the same spot at the top and on the list afterwards
